pytensor/tensor/extra_ops.py

```diff
@@ -593,12 +593,12 @@ def squeeze(x, axis=None):
         # Nothing to do
         return _x
 
-    axes = ()
-    for i in axis:
-        if i < _x.ndim and not _x.broadcastable[i]:
-            axes += (i,)
-    if axes:
-        _x = specify_broadcastable(_x, axes)
+    # if the axis is of length 1 but is not specifed as broadcastable, we first explicitly specify it as broadcastable
+    non_broadcastable_axis = [
+        i for i in axis if i < _x.ndim and not _x.broadcastable[i]
+    ]
```

Comment:
We can handle the `_x.ndim` case separately, and then you don't need that first check:

```suggestion
    if _x.ndim == 0:
        # Nothing could be squeezed
        return _x
        
    non_broadcastable_axis = [
        i for i in axis if not _x.broadcastable[i]
    ]
```